pkg/docker/docker.go

```diff
@@ -800,15 +926,18 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 			// the container exits normally).  dump port/etc information for the user.
 			dumpContainerInfo(container, d, image)
 		}
+
 		// Return an error if the exit code of the container is
 		// non-zero.
 		glog.V(4).Infof("Waiting for container %q to stop ...", containerName)
-		exitCode, err := d.client.WaitContainer(container.ID)
+		ctx, cancel := getDefaultContext(15 * time.Minute)
```

Comment:
I think we should retain the existing behavior (wait forever). If we hardcode the build to a 15min timeout, no one is going to be able to build anything that takes longer than 15 mins, which is clearly unacceptable, and while we could add a --timeout argument to s2i, we'd then also have to add that field to openshift which again, seems undesirable.

for run it definitely makes sense to wait forever.